src/App.js

```diff
@@ -362,57 +426,65 @@ function App() {
                     });
                 });
 
-                batch.commit();
-                setSubcollectionName(undo.oldList);
-            } else {
-                onEditList(undo.newList, undo.oldList);
+                setListInfo({
+                    listId: undo.oldListId,
+                    listName: undo.oldListName,
+                });
             }
+
+            batch.commit();
         }
 
+
         // Update the undoStack
         setUndoStack(newStack);
     }
 
     return (
         <div className="App">
-            {(itemsLoading || userLoading) && <div className="loading-spinner"></div>}
-            {(itemsError || userError) && <h1 className="empty-placeholder">Error occurred while trying to fetch data. Please try again later.</h1>}
-            {!itemsLoading && !userLoading && !itemsError && !userError &&
+            {(listsLoading || itemsLoading) && <div className="loading-spinner"></div>}
+            {(listsError || itemsError) && <h1 className="empty-placeholder">Error occurred while trying to fetch data. Please try again later.</h1>}
+            {!listsLoading && !itemsLoading && !listsError && !itemsError &&
                 <>
                     {/* Top title bar with list changing buttons/dropdown */}
                     <div className="todo-text todo-title">
                         <div className="todo-list-dropdown-container">
                             <div className="todo-list-dropdown">
-                                {/* To-Do: */}
+                                <span class="todo-list-dropdown-label">To-Do:</span>
                                 <Dropdown
-                                    menuLabel="To-Do:"
+                                    selectClass={"main-list-select-mui"}
+                                    dropdownWidth={200}
+                                    menuLabel=""
                                     onSelectItem={(val) => {
```

Comment:
This is fragile. There may be multiple lists with the same name.  Have onSelectItem provide the ID rather than the name.

Reply:
Good point. We noticed this when working on the next part, so we fixed it to have the display option be the list name but the actual value be the list ID.